### PR TITLE
Reduced Cognitive complexity of the code.

### PR DESCRIPTION
**Pull request by anjaliInfy1822**
Reduced the cognitive complexity of the code, by reducing the if else and for loop nesting multiple times.

### CODE REVIEW

File: catalog/camel-catalog/src/main/java/org/apache/camel/catalog/DefaultCamelCatalog.java [hunk 1]
```diff
@@ -659,4 +654,15 @@ private String loadResource(String file) {
         }
     }
 
+    private ArtifactModel matchArtifactModelFromList(List<String> artifactModelList, String groupId, String artifactId,
+            String version) {
+        for (String name : artifactModelList) {
+            ArtifactModel<?> am = componentModel(name);
```
Comment (davsclaus) [inline catalog/camel-catalog/src/main/java/org/apache/camel/catalog/DefaultCamelCatalog.java:660]: this is wrong, here is only component, the old had per different type

**Review by davsclaus — CHANGES_REQUESTED** [review]

File: catalog/camel-catalog/src/main/java/org/apache/camel/catalog/DefaultCamelCatalog.java (hunk 1 above)
Comment (anjaliInfy1822) [reply catalog/camel-catalog/src/main/java/org/apache/camel/catalog/DefaultCamelCatalog.java:660]: Hi @davsclaus , sorry I missed that part. Would it be okay to have different function for each case?

Comment (davsclaus) [reply catalog/camel-catalog/src/main/java/org/apache/camel/catalog/DefaultCamelCatalog.java:660]: This PR does not improve anything, the old code is fine, and also avoids backport problems.

If you want to help with Camel, then help with better things, than this stuff. Also it helps to get involved with Camel first than open your first PR out of "the blue".

Comment (anjaliInfy1822) [reply catalog/camel-catalog/src/main/java/org/apache/camel/catalog/DefaultCamelCatalog.java:661]: @davsclaus  sure Thanks

Comment (orpiske) [reply catalog/camel-catalog/src/main/java/org/apache/camel/catalog/DefaultCamelCatalog.java:661]: @anjaliInfy1822 My .2 cents: focus on the "easy" tickets and "help wanted" first to gain some experience with the code base before proceeding with things like this. 
As Claus said, get involved with the community (specially on the Zulip chat) to keep an eye on things we need help with.
Once you have a good grasp of the code base and know what to expect, then it's OK to focus on cleaning up things (we do have a lot of things to cleanup!!! But this one is not one of them).

**Pull request closed without merging**